src/tools/compiletest/src/runtest.rs

```diff
@@ -3068,14 +3068,24 @@ impl<'test> TestCx<'test> {
 
     fn check_mir_dump(&self) {
         let test_file_contents = fs::read_to_string(&self.testpaths.file).unwrap();
+        let target = if self.config.target.contains("x86_64") {
+            "x86_64"
+        } else if self.config.target.contains("i686") {
+            "i686"
+        } else {
+            // Don't run output tests on other platforms
+            return;
+        };
```

Comment:
This is very different from what I had in mind, which is to use *revisions* (an existing feature of `compiletest`) inside specific tests to test both 32-bit and 64-bit targets.

Revisions already suffix the files appropriately so you wouldn't need separate directories.

But also, I very much want to see this use `32`/`64` (maybe `32-bit` and `64-bit`) names and not care about the CPU architecture, but rather about the pointer size.

Reply:
revisions won't work. all platforms run all revisions. what you are suggesting needs a new system, this is the hacky protoype of such a system

Reply:
> all platforms run all revisions

Naively I would expect `//[64-bit] only-x86_64` to work.
And then replace `x86_64` with something that applies to all 64-bit targets.

Reply:
That's horrible. And not how revisions are meant to be used. They are for different variants of the same file, not running the same file on different platforms, but only when targetting that platform. If people want actual revisions in a mir-opt test, then it'll explode in complexity, either making ppl not use revisions where they would apply, or causing mistakes. Let's find a simple system (e.g. the current one) that isn't easy to get wrong.

Reply:
Fair enough, but can it be opt-in, so we don't do it for tests that don't need it?

Reply:
That already works. If you specify that the test should only be run on 64 bit we don't attempt to produce MIR for it

Reply:
Uh, no, I mean tests that have the same output on all platforms.

I want to keep the number of tests that don't run on some platforms at an absolute minimum. They're potentially hiding catastrophic bugs.

Reply:
It's opt-in now